mlrun/api/crud/model_monitoring/_model_endpoint_stores/kv_model_endpoint_store.py

```diff
@@ -0,0 +1,422 @@
+            )
+
+        # Cleanup TSDB
+        frames = mlrun.utils.v3io_clients.get_frames_client(
+            token=self.access_key,
+            address=mlrun.mlconf.v3io_framesd,
+            container=self.container,
+        )
+
+        # Generate the required tsdb paths
+        tsdb_path, filtered_path = self._generate_tsdb_paths()
+
+        # Delete time series DB resources
+        try:
+            frames.delete(
+                backend=model_monitoring_constants.TimeSeriesTarget.TSDB,
+                table=filtered_path,
+                if_missing=v3io_frames.frames_pb2.IGNORE,
+            )
+        except v3io_frames.errors.CreateError:
+            # Frames might raise an exception if schema file does not exist.
+            pass
+
+        # Final cleanup of tsdb path
+        tsdb_path.replace("://u", ":///u")
+        store, _ = mlrun.store_manager.get_or_create_store(tsdb_path)
+        store.rm(tsdb_path, recursive=True)
+
+    def _generate_tsdb_paths(self) -> typing.Tuple[str, str]:
+        """Generate a short path to the TSDB resources and a filtered path for the frames object
+        :return: A tuple of:
+             [0] = Short path to the TSDB resources
+             [1] = Filtered path to TSDB events without schema and container
+        """
+        # Full path for the time series DB events
+        full_path = (
+            mlrun.mlconf.model_endpoint_monitoring.store_prefixes.default.format(
+                project=self.project,
+                kind=mlrun.api.schemas.ModelMonitoringStoreKinds.EVENTS,
+            )
+        )
+
+        # Generate the main directory with the TSDB resources
+        tsdb_path = mlrun.utils.model_monitoring.parse_model_endpoint_project_prefix(
+            full_path, self.project
+        )
+
+        # Generate filtered path without schema and container as required by the frames object
+        (
+            _,
+            _,
+            filtered_path,
+        ) = mlrun.utils.model_monitoring.parse_model_endpoint_store_prefix(full_path)
+        return tsdb_path, filtered_path
+
+    @staticmethod
+    def build_kv_cursor_filter_expression(
+        project: str,
+        function: str = None,
+        model: str = None,
+        labels: typing.List[str] = None,
+        top_level: bool = False,
+    ) -> str:
+        """
+        Convert the provided filters into a valid filter expression. The expected filter expression includes different
+        conditions, divided by ' AND '.
+
+        :param project:    The name of the project.
+        :param model:      The name of the model to filter by.
+        :param function:   The name of the function to filter by.
+        :param labels:     A list of labels to filter by. Label filters work by either filtering a specific value of
+                           a label (i.e. list("key==value")) or by looking for the existence of a given
+                           key (i.e. "key").
```

Comment:
For the KV filter query we already have this function (`build_kv_cursor_filter_expression()`) that parse that list of labels into a valid KV filter query. Before we pass the labels into that function, we convert the labels **dictionary** into a **list**. 
Anyway, in a different PR, I will make changes in this `build_kv_cursor_filter_expression()` function so it will handle the labels dictionary instead of the labels list.